bolt-simple-taproot.md

```diff
@@ -377,348 +352,276 @@ The Context column decodes as follows:
+- MUST compute the aggregate nonce from:
+  - the `musig2_nonce` field the recipient previously sent in `funding_created`
+  - the `public_nonce` included as part of the `partial_signature_with_nonce` field
+- MUST verify the `partial_signature_with_nonce` field using the `PartialSigVerifyInternal`
+  algorithm of `bip-musig2`:
+  - if the partial signature is invalid, MUST fail the channel
 
-We add a new TLV field to the `funding_locked` message:
+#### `channel_ready`/`channel_reestablish` Extensions
 
-1. type: 36 (`funding_locked`)
-2. data:
-    * ...
-    * [`funding_locked_tlvs`:`tlvs`]
+We add a new TLV field to the `channel_ready` and `channel_reestablish` messages:
 
-1. `tlv_stream`: `funding_locked_tlvs`
+1. `tlv_stream`: `channel_ready_tlvs`/`channel_reestablish_tlvs`
 2. types:
-    1. type: 0 (`verification_musig2_pubnonce`)
-    2. data:
-        * [`musig2_pubnonce`:`nonces`]
-    1. type: 2 (`signing_musig2_pubnonce`)
-    2. data:
-        * [`musig2_pubnonce`:`nonces`]
-
-Similar to the `next_per_commitment_point`, by sending the `musig2_pubnonce`
-value in this message, we ensure that the remote party has our public nonce
-which is required to generate a new commitment signature. Similarly, we can't
-generate a new signature until we receive their `verification_musig2_pubnonce`
-and `signing_musig2_pubnonce`.
+   1. type: 4 (`musig2_nonce`)
+   2. data:
+      * [`66*byte`: `public_nonce`]
 
-##### Requirements
+Similar to the `next_per_commitment_point`, by sending the `musig2_nonce` value in this message,
+we ensure that the remote party has our public nonce which is required to generate a new
+commitment signature.
 
-The sender MUST:
+##### Requirements
 
- - MUST set `verification_musig2_pubnonce` and `signing_musig2_pubnonce` to a
-   _fresh_ set of public nonces as specified by `bip-musig2`
+The sender:
 
-The receiver MUST:
+- MUST set `musig2_nonce` to a fresh, unique `musig2` nonce as specified by `bip-musig2`
 
-  - MUST combine their local `pubnonce` and the received `pubnonce` via the
-    `NonceAgg` algorithm specified in the `musig2` BIP to received _two_
-    combined nonces. This combined nonce will be used to generate the next
-    commitment signature via the `Sign` algorithm specified in the `musig2`
-    BIP.
+The recipient:
 
+- MUST fail the channel if `musig2_nonce` is absent.
+- MUST store `musig2_nonce` until they send the next `commitment_signed` message.
+  - SHOULD only store `musig2_nonce` in RAM, and not persist it to disk.
 
 ### Cooperative Closure
 
-#### `shutdown` Extensions
-
-A new TLV field is added to the `shutdown` message:
+#### `closing_signed` Extensions
 
-1. type: 38 (`shutdown`)
-2. data:
-    * ...
-    * [`shutdown_tlvs`:`tlvs`]
+We add a new TLV to the `closing_signed`
+message's existing `tlv_stream`:
 
-1. `tlv_stream`: `shutdown_tlvs`
+1. `tlv_stream`: `closing_signed_tlvs`
 2. types:
-    1. type: 2 (`musig2_pubnonce`)
-    2: data:
-        * [`66*byte`:`nonces`]
```

Comment:
> The nice thing about sending it in the shutdown message is that you effectively create an entirely new musig2 session. Otherwise, you'll need to thread through that last living session into w/e handles the co-op close state machine.

What do you think about this point? If we send in `shutdown`, then w/e handles that state machine can generate its own nonces. Otherwise the nonces need to be passed down thru w/e layers/connections, which may potentially require stronger synchronization.